node/pkg/db/rediscriber.go

```diff
@@ -0,0 +1,112 @@
+package db
+
+import (
+	"context"
+
+	errorsentinel "bisonai.com/orakl/node/pkg/error"
+	"github.com/redis/go-redis/v9"
+	"github.com/rs/zerolog/log"
+)
+
+type Rediscriber struct {
+	client        *redis.Client
+	router        func(*redis.Message) error
+	redisChannels []string
+}
+
+type RediscriberConfig struct {
+	RedisHost     string
+	RedisPort     string
+	RedisChannels []string
+	Router        func(*redis.Message) error
+}
+
+type RediscriberOption func(*RediscriberConfig)
+
+func WithRedisHost(host string) RediscriberOption {
+	return func(config *RediscriberConfig) {
+		config.RedisHost = host
+	}
+}
+
+func WithRedisPort(port string) RediscriberOption {
+	return func(config *RediscriberConfig) {
+		config.RedisPort = port
+	}
+}
+
+func WithRedisChannels(channels []string) RediscriberOption {
+	return func(config *RediscriberConfig) {
+		config.RedisChannels = channels
+	}
+}
+
+func WithRedisRouter(router func(*redis.Message) error) RediscriberOption {
+	return func(config *RediscriberConfig) {
+		config.Router = router
+	}
+}
+
+func NewRediscriber(ctx context.Context, opts ...RediscriberOption) (*Rediscriber, error) {
+	config := &RediscriberConfig{}
+	for _, opt := range opts {
+		opt(config)
+	}
+
+	if config.Router == nil {
+		return nil, errorsentinel.ErrRediscriberRouterNotFound
+	}
+
+	client, err := newRedisClient(ctx, config.RedisHost, config.RedisPort)
+	if err != nil {
+		return nil, err
+	}
+	return &Rediscriber{
+		client:        client,
+		router:        config.Router,
+		redisChannels: config.RedisChannels,
+	}, nil
+}
+
+func (r *Rediscriber) Start(ctx context.Context) error {
+	err := r.client.Ping(ctx).Err()
```

Comment:
is this same check performed in `newRedisClient`? If so, I suggest removing one

Reply:
or is it intentional in case if redis goes down after being initialized? if so, i think it's okay to keep 😄